[variant] Add Generic F4xx

ABOSTM opened the pull request:
**Summary**
[variant] Add Generic F4xx

Supersede #1337

gbmhunter committed:
Added NUCLEO_F070RB

Supersede #1167

Signed-off-by: Geoffrey Hunter <gbmhunter@gmail.com>
Co-authored-by: Frederic.Pillon <frederic.pillon@st.com>

jhleekr committed:
Add Aurora One Variant

  https://www.bfy.kr/aurora-one/

Supersede #1176

Signed-off-by: JongHyeon Lee <jhlee@bfy.kr>
Co-authored-by: Frederic.Pillon <frederic.pillon@st.com>

vladkozlov69 committed:
Added Generic H743Vx

Supersede #1192

Signed-off-by: Vladimir Kozlov <vkozloff@gmail.com>
Co-authored-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
[variant] Update Generic F0xx

All specific F0 variants moved to the generic variant.

Supersede  #1080

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
[variant] Update Generic MP1xx

As STM32MP1 is particular, the generic target have to be customized
by end user based on the used device tree.
So, currently, generic menu will not be deployed.

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
[stm2variant] Add board_entry.txt

This will help to add generic board entry.
Note:
upload.maximum_size and product_line have to be verified
and changed if needed.

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
[stm32variant] Add option to generate only one family

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
Allow use of a custom peripheral pins file

Syntax in the board.txt:
xxx.build.peripheral_pins=-DCUSTOM_PERIPHERAL_PINS

Example to use a custom peripheral pins file for Nucleo_L476RG:
    Nucleo_64.menu.pnum.NUCLEO_L476RG.build.peripheral_pins=-DCUSTOM_PERIPHERAL_PINS
Then add a file with ".c" extension like "PeripheralPins_<board>.c" in the
variantfolder.

At sketch folder:
CUSTOM_PERIPHERAL_PINS could now be overridden using the build_opt.h, ex:
-DCUSTOM_PERIPHERAL_PINS=\"PeripheralPins_NUCLEO_L476RG.c\"

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
[stm32variant] Add generic_clock.c generation

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
[stm32variant] Aggregate generated variant if they are the same

To avoid having duplicated variants with all the same files.

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
[stm32variant] Comment option to generate from one mcu file

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
[CI] arduino-builder is replaced by arduino-cli

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

seeers committed:
Add Generic F042Cx

Supersede #1103

Signed-off-by: seeers <seeers@gmx.de>
Co-authored-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
[stm32variant] Change generic variant generation

- variant.* files are renamed variant_generic.*
Generic variant header file will be included thanks
a single variant.h using a variable defined in boards.txt
- Clean template to remove useless extern "C"
- Include guards replaced by #pragma once directive

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
Update variant header management

By default the variant.h will include variant_generic.h file.
*.build.variant_h=variant_{build.board}.h can be added in
the boards.txt to define the dedicated header to a board.

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
[PIO]Fix variant header changes

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
[variant] Add Generic L4xx

All specific L4 variants moved to the generic variant.

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
[variant] Add Generic F2xx

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
[variant] Add Generic L5xx and Nucleo L552ZE-Q

Supersede #1256

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
[variant] Add Generic WBxx

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
Add all generated STM32F0xx generic variant files

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
Add all generated STM32F1xx generic variant files

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
Add all generated STM32F2xx generic variant files

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
Add all generated STM32F3xx generic variant files

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
Add all generated STM32F4xx generic variant files

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
Add all generated STM32F7xx generic variant files

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
Add all generated STM32G0xx generic variant files

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
Add all generated STM32G4xx generic variant files

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
Add all generated STM32H7xx generic variant files

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
Add all generated STM32L0xx generic variant files

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
Add all generated STM32L1xx generic variant files

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
Add all generated STM32L4xx generic variant files

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
Add all generated STM32L5xx generic variant files

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
Add all generated STM32MP1xx generic variant files

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
Add all generated STM32WBxx generic variant files

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
Add all generated STM32WLxx generic variant files

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

ABOSTM committed:
[variant] Add Generic G4xx

All specific G4 variants moved to the generic variant.

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

fpistm committed:
[variant] Add Generic G0xx

Nucleo G071RB and Disco G0316 moved to the generic variant.

Fix #857

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
Add Nucleo G031K8 and generic G031Kx

Fix #855

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
[variant] Add Generic L1xx

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
[variant] Update Generic H7xx

Fix #1276 and Fix #1277

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
Enable HAL MDMA by default

Some series require to have it defined by default (H7, MP1)

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
[CI] Update configuration to build only one generic per variant

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

ABOSTM committed:
[variant] Add Generic F7xx

Fix #1302

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
[variant] Add Generic L0xx

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

owennewo committed:
[variant] Add support of board B-G431B-ESC1

Supersede #1236

Signed-off-by: Owen Williams <williams.owen@gmail.com>
Co-authored-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

mkals committed:
Add Generic L010RB

Supersede #1123

Signed-off-by: Morten Kals <morten@kals.no>
Co-authored-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
[variant] Add Generic F3xx

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

Stanimir-Petev committed:
[Variant] Add OLIMEXINO-STM32F3

Supersede #1115

Signed-off-by: Stanimir Petev <stanimir.petev@gmail.com>
Co-authored-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

dnandha committed:
[Variant] Add DISCO F303VC

Supersede #1238

Signed-off-by: dnandha <daljeet@gmx.net>
Co-authored-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
[variant] Add Generic F4xx

Supersede #1337

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

fpistm committed:
[stm32variant] Fix USB mode when AF is not none

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

ABOSTM committed:
BlackPill F401CC: add USER_BTN on PA0

Note this user buuton is not available on all revisions of this board

Fixes #1144

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

dds90 committed:
[New variant] STM32F407ZGT6 Mini

Add [New variant] STM32F407ZGT6 Mini and Generic_F407Zx
Supersed #1130

Signed-off-by: dds90 <40141422+dds90@users.noreply.github.com>
Co-authored-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

fpistm commented:
One point raised on the forum which could be fixed in this PR
https://www.stm32duino.com/viewtopic.php?p=6652#p6652

ABOSTM committed:
BLACK_F407ZX: invert PF9 and PF10 in digitalPin[]

Invert PF9 and PF10 in digitalPin[] to match their definition in
variant.h
Fixes forum issue:
https://www.stm32duino.com/viewtopic.php?p=6652#p6652

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

fpistm commented:
Integrated to official PR.

Thanks
